src/prebundle-modules-plugin.ts

```diff
@@ -0,0 +1,180 @@
+import esbuildCommonjsPlugin from '@chialab/esbuild-plugin-commonjs';
+import { stringifyProcessEnvs } from '@storybook/core-common';
+import type { Plugin } from '@web/dev-server-core';
+import { build } from 'esbuild';
+import { join } from 'path';
+import { getNodeModuleDir } from './get-node-module-dir';
+
+export const PREBUNDLED_MODULES_DIR = 'node_modules/.prebundled_modules';
+
+export function prebundleModulesPlugin(env: Record<string, string>): Plugin {
+  const modulePaths: Record<string, string> = {};
+
+  return {
+    name: 'prebundle-modules',
```

Comment:
we should add a task/story to not prebundle if the user still has a valid and up to date `.prebundled_modules` already